### DISCUSSION TITLE
React Hook Form - useFieldArray as separate component

### DISCUSSION THREAD

**Discussion by tanetrail** [react-hook-form/react-hook-form#3846] [Q&A] [answered]
I created a react component with useFieldArray and tried to access it from another react component. when i tried this approach i cant populate the default values and also the delete is not working. any idea what went wrong in my code. below is the sample application.
https://stackblitz.com/edit/ionic-react-hook-form-xwvlf9?file=src%2FAppUtil.tsx

**Comment by bluebill1049** [comment, marked as answer]
you having those inner component which re-mount and unmount itself consistently.

**Reply by tanetrail** [reply]
Could you please suggest an option to manage the field array as a separate component

**Reply by bluebill1049** [reply]
this helps? https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr
